src/createListComponent.js

```diff
@@ -171,14 +171,20 @@ export default function createListComponent({
     }
 
     scrollTo(scrollOffset: number): void {
+      const element = ((this._outerRef: any): HTMLDivElement);
+      this._scrollCorrectionInProgress = true;
       this.setState(
         prevState => ({
           scrollDirection:
             prevState.scrollOffset >= scrollOffset ? 'backward' : 'forward',
           scrollOffset: scrollOffset,
           scrollUpdateWasRequested: true,
         }),
-        this._resetIsScrollingDebounced
+        () => {
```

Comment:
@enahum This is the change i was referring to for fixing position when there are lot of unread.

Reply:
Ok, still not following what has to be done. Also saw the comment from @saturnino about the permalink view and wonder how will that affect

Reply:
I removed the scrollOffset correction on componentDidUpdate as the flag `scrollUpdateWasRequested ` will be true and it triggers that offset correction causing issues. 

I am guessing this should correct it but nevertheless this makes the code predictable as this is the flow for which we need that correction in componentDidMount.

Whenever we update position with scrollTo we set the state value of the offset here and correct the scroll position in componentDidUpdate using the flag `scrollUpdateWasRequested ` but as this flag is not reset unless user scrolls we might actually exec that scroll correction code as side effect for other scenarios.